receiver/googlecloudpubsubpushreceiver/factory.go

```diff
@@ -5,14 +5,18 @@ package googlecloudpubsubpushreceiver // import "github.com/open-telemetry/opent
 
 import (
 	"context"
+	"errors"
 
 	"go.opentelemetry.io/collector/component"
 	"go.opentelemetry.io/collector/consumer"
 	"go.opentelemetry.io/collector/receiver"
 
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/sharedcomponent"
 	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/googlecloudpubsubpushreceiver/internal/metadata"
 )
 
+var receivers = sharedcomponent.NewSharedComponents()
```

Comment:
I have added them to a shared receivers map @axw

Reply:
you only need this if you reuse the same component across signals.